File: src/main/java/org/jdc/template/model/database/main/individualquery/IndividualQuery.java
/*
 * IndividualQuery.java
 *
 * Created: 09/17/2014 03:15:27
 */



package org.jdc.template.model.database.main.individualquery;

import android.database.Cursor;
import android.util.Log;

import org.dbtools.android.domain.database.contentvalues.DBToolsContentValues;
import org.dbtools.query.sql.SQLQueryBuilder;
import org.jdc.template.model.database.main.individual.IndividualConst;

public class IndividualQuery extends IndividualQueryBaseRecord {

    public static final String QUERY;

    static {

        QUERY = new SQLQueryBuilder()
                .field(IndividualConst.FULL_C_ID, IndividualQueryConst.C_ID)
                .field(IndividualConst.FULL_C_FIRST_NAME, IndividualQueryConst.C_NAME)
                .table(IndividualConst.TABLE)
                .toString();

        Log.e("ldst", "QUERY: " + QUERY);
    }

    public IndividualQuery(IndividualQuery record) {
        super(record);
    }

    public IndividualQuery(Cursor cursor) {
        setContent(cursor);
    }

    public IndividualQuery(DBToolsContentValues values) {
        setContent(values);
    }

    public IndividualQuery() {
    }


}

File: src/main/kotlin/org/jdc/template/model/database/main/individualquery/IndividualQuery.kt
/*
 * IndividualQuery.kt
 *
 * Created: 02/14/2016 07:02:10
 */



package org.jdc.template.model.database.main.individualquery

import android.database.Cursor
import org.dbtools.android.domain.database.contentvalues.DBToolsContentValues
import org.dbtools.query.sql.SQLQueryBuilder

class IndividualQuery : IndividualQueryBaseRecord {

    companion object {
        val QUERY: String
        init {
            QUERY = SQLQueryBuilder()
                    .field(IndividualQueryConst.FULL_C_ID, IndividualQueryConst.C_ID)
                    .field(IndividualQueryConst.FULL_C_NAME, IndividualQueryConst.C_NAME)
                    .table("FROM SOME TABLE(S)")
                    .buildQuery();

        }
    }

    constructor() {
    }

    constructor(record: IndividualQuery) : super(record) {
    }

    constructor(cursor: Cursor) {
        setContent(cursor)
    }

    constructor(values: DBToolsContentValues<*>) {
        setContent(values)
    }


}